float32[] numbers #n
bool[] flags #b
uint8 msg_id